### PR TITLE
ocf_desktop: replace ocf-suspend using ocflib

### PR DESCRIPTION
**Pull request by chriskuehl**
Benefits of the change (in decending order of importance):

  * Uses ocflib hours, so we no longer will have desktops running over breaks when we forget to disable wakeups.

  * Adds some output that can be useful for determining why a desktop isn't suspending regularly.

  * Python

There are probably a few bugs in this that we'll have to fix, but hopefully this is an improvement.

### CODE REVIEW

File: modules/ocf_desktop/files/suspend/ocf-suspend [hunk 1]
```diff
@@ -1,28 +1,85 @@
...
+
+from ocflib.lab.hours import Day
+
+
+devnull = open('/dev/null', 'w')
+
+
+def file_is_open(path):
+    """Return whether a file is open by any process on the system."""
+    return subprocess.call(('lsof', '--', path), stdout=devnull, stderr=devnull) == 0
```
Comment (kpengboy) [inline modules/ocf_desktop/files/suspend/ocf-suspend]: You can use `subprocess.DEVNULL` instead

Comment (chriskuehl) [inline modules/ocf_desktop/files/suspend/ocf-suspend]: good catch, forgot we're on python3.4 now
added in a new commit

File: modules/ocf_desktop/files/suspend/ocf-suspend [hunk 2]
```diff
@@ -1,28 +1,83 @@
...
+
+def main(argv=None):
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument('--force', '-f', action='store_true', help='force suspend, even if the lab is open')
+    parser.add_argument('--quiet', '-q', action='store_true', help='stay quiet except for unexpected errors')
+    args = parser.parse_args(argv)
+
+    def report(line):
+        if not args.quiet:
+            print('[ocf-suspend] {}'.format(line), file=sys.stderr)
```
Comment (kpengboy) [inline modules/ocf_desktop/files/suspend/ocf-suspend]: ...wait, do we have other commands doing it this way? Most programs I've seen use the `programname: errormessage` format

Comment (chriskuehl) [inline modules/ocf_desktop/files/suspend/ocf-suspend]: shrug

**Comment by kpengboy** [comment]
lg2m, but maybe I missed something

**Comment by chriskuehl** [comment]
thanks for quick review

i'm guessing there are still some bugs (or at least some commands that can fail and cause cron spam that we don't want), so we'll just iterate on this until it's working...

File: modules/ocf_dhcp/manifests/init.pp [hunk 3]
```diff
@@ -30,27 +30,18 @@
...
-      minute  => 0,
-      weekday => 6,
-      require => File['/usr/local/bin/lab-wakeup'];
-    'lab-wakeup-sunday':
-      command => '/usr/local/bin/lab-wakeup > /dev/null',
-      hour    => 12,
-      minute  => 0,
-      weekday => 0,
+    'lab-wakeup':
+      command => '/usr/local/bin/lab-wakeup',
```
Comment (kpengboy) [inline modules/ocf_dhcp/manifests/init.pp]: You meant `lab-wakeup -q`.

**Pull request merged**